spec/lib/psulib_traject/processors/call_number/dewey_spec.rb

```diff
@@ -3,8 +3,135 @@
+        end
+
+        it 'allows for characters after the number in the doon (e.g. 12TH)' do
+          expect(described_class.new('123.23 20TH .M45 V.1').doon1).to match(/^20TH/)
+        end
+      end
+
+      describe 'cutters' do
+        it 'handles 3 possible cutters' do
+          expect(described_class.new('123.23 .M23 V.1 2002').cutter1).to eq '.M23'
+          expect(described_class.new('123.23 .M23 V.1 2002').cutter2).to be_nil
+          expect(described_class.new('123.23 .M23 V.1 2002').cutter3).to be_nil
+
+          expect(described_class.new('123.23 .M23 .M45 V.1 2002').cutter1).to eq '.M23'
+          expect(described_class.new('123.23 .M23 .M45 V.1 2002').cutter2).to eq '.M45'
+          expect(described_class.new('123.23 .M23 .M45 V.1 2002').cutter3).to be_nil
+
+          expect(described_class.new('123.23 .M23 .M45 .S32 V.1 2002').cutter1).to eq '.M23'
+          expect(described_class.new('123.23 .M23 .M45 .S32 V.1 2002').cutter2).to eq '.M45'
+          expect(described_class.new('123.23 .M23 .M45 .S32 V.1 2002').cutter3).to eq '.S32'
+        end
+
+        it 'handles multi-letter cutters' do
+          expect(described_class.new('123.23 .MS23').cutter1).to eq '.MS23'
+          expect(described_class.new('123.23 .MSA23').cutter1).to eq '.MSA23'
+        end
+
+        it 'handles a multiple letters after the cutter number' do
+          expect(described_class.new('123.23 .MS23ABC').cutter1).to eq '.MS23ABC'
+        end
+
+        it 'parses cutters that do not have a space before them' do
+          expect(described_class.new('123.23.M23.S32').cutter1).to eq '.M23'
+          expect(described_class.new('123.23.M23.S32').cutter2).to eq '.S32'
+        end
+
+        it 'parses cutters delimited by a slash' do
+          expect(described_class.new('123.23/M23/S32').cutter1).to eq '/M23'
+          expect(described_class.new('123.23/M23/S32').cutter2).to eq '/S32'
+        end
+
+        it 'parses cutters with no space or period' do
+          expect(described_class.new('123M23S').cutter1).to eq 'M23S'
+          expect(described_class.new('123M23S32').cutter1).to eq 'M23'
+          expect(described_class.new('123M23S32').cutter2).to eq 'S32'
+        end
+      end
+
+      describe 'the rest of the stuff' do
+        it 'puts any other content into the rest attribute' do
+          expect(described_class.new('123.23 .M23 V.1 2002').rest).to eq 'V.1 2002'
+          expect(described_class.new('123.23 2012 .M23 2002 .M45 V.1 2002-2012/GobbldyGoop').rest)
+            .to eq 'V.1 2002-2012/GobbldyGoop'
+        end
+      end
+    end
+  end
+
+  describe '#reduce' do
+    RSpec::Matchers.define :reduce_to do |expected|
+      match do |actual|
+        described_class.new(actual).reduce == expected
+      end
+      failure_message do |actual|
+        "expected #{actual} to reduce to #{expected} but got #{described_class.new(actual).reduce} instead."
+      end
+    end
```

Comment:
If we like these, we could move them out to their own file in `spec/support` to share with the other spec tests.

Reply:
yup that is what I was thinking too, can move that in this PR

Reply:
updated